datafusion/core/src/physical_plan/projection.rs

```diff
@@ -410,12 +415,39 @@ fn stats_projection(
             .collect()
     });
 
-    Statistics {
-        is_exact: stats.is_exact,
-        num_rows: stats.num_rows,
-        column_statistics,
-        // TODO stats: knowing the type of the new columns we can guess the output size
-        total_byte_size: None,
+    let primitive_row_size = inner_exprs
+        .clone()
+        .into_iter()
+        .map(|e| match e.data_type(schema.as_ref()) {
+            Ok(data_type) => data_type.primitive_width(),
+            Err(_) => None,
+        })
+        .fold(Some(0usize), |init, v| match (init, v) {
+            (Some(l), Some(r)) => Some(l + r),
+            _ => None,
+        });
+
+    match (primitive_row_size, stats.num_rows) {
+        (Some(row_size), Some(row_count)) => {
+            Statistics {
+                is_exact: stats.is_exact,
+                num_rows: stats.num_rows,
+                column_statistics,
+                // Use the row_size * row_count as the total byte size
+                total_byte_size: Some(row_size * row_count),
+            }
+        }
+        _ => {
+            Statistics {
+                is_exact: stats.is_exact,
+                num_rows: stats.num_rows,
+                column_statistics,
+                // TODO stats: knowing the type of the new columns we can guess the output size
+                // If we can't get the exact statistics for the project
+                // Before we get the exact result, we just use the child status
+                total_byte_size: stats.total_byte_size,
```

Comment:
```suggestion
                total_byte_size: None,
```

Reply:
Do you means we use the `None` or `unknow` as the `total_byte_size`, if we can't estimate the `total_byte_size`?
But it will lost the `size` information in side, I mean use the child's statistics directly.

cc @jackwener

Reply:
Yes, because it will cause inaccuracy to use `child's statistics` directly.

Reply:
I think we might need to add exactness information (is_exact) to `total_byte_size` as well to prevent not propagating some (inaccurate) information about the size.

Reply:
I think it is fine to leave it inaccurate for now, as AFAIK we don't rely on exact information about `total_byte_size` like we do for `num_rows` statistics.

Reply:
cc @Dandandan @jackwener any comments for this
If there is no comments for this, i will merged this pr after 24h.

Reply:
LGTM❤️